### PR TITLE
KAFKA-14774 the removed listeners should not be reconfigurable

### PR DESCRIPTION
**Pull request by chia7712**
Users can alter broker configuration to remove specify listeners. However, the removed listeners are NOT removed from `reconfigurables` list. It can result in the idle processors if users increases the network threads subsequently.

the following screenshot show a example of increasing network thread from 8 to 10, and we still create threads for the removed listener (`PLAINTEXT2`)

截圖 2023-03-21 下午7 16 33

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@showuon @mumrah Could you take a look?

File: core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala [hunk 1]
```diff
@@ -1167,6 +1168,35 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
     assertTrue(partitions.exists(_.leader == null), "Did not find partitions with no leader")
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def test(quorum: String): Unit = {
```
Comment (mimaison) [inline core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala]: Can we find a better name?

**Review by mimaison — COMMENTED** [review]
Nice find! Changes look good, we just need to rename the test :)

**Comment by chia7712** [comment]
@mimaison thanks for feedback. Have better naming in latest commit.

**Review by mimaison — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
failed test is traced by #13452

**Comment by showuon** [comment]
Nice find @chia7712 , I think we should back-port this to 3.4 branch. WDYT?

**Comment by chia7712** [comment]
> I think we should back-port this to 3.4 branch. WDYT?

Will copy that

**Pull request merged**